src/WebJobs.Script/Binding/HttpBinding.cs

```diff
@@ -31,88 +34,105 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
+            }
+            else
+            {
+                // Handle ExpandoObjects
+                responseObject = content as IDictionary<string, object>;
+            }
+
             HttpStatusCode statusCode = HttpStatusCode.OK;
-            JObject headers = null;
+            IDictionary<string, object> responseHeaders = null;
             bool isRawResponse = false;
-            if (content is string)
+            if (responseObject != null)
             {
-                try
-                {
-                    // attempt to read the content as a JObject
-                    JObject jo = JObject.Parse((string)content);
-
-                    // if the content is json we capture that so it will be
-                    // serialized as json by WebApi below
-                    content = jo;
-
-                    // TODO: Improve this logic
-                    // Sniff the object to see if it looks like a response object
-                    // by convention
-                    JToken value = null;
-                    if (jo.TryGetValue("body", StringComparison.OrdinalIgnoreCase, out value))
-                    {
-                        content = value;
+                responseObject = responseObject.AsCaseInsensitive();
 
-                        if (value is JValue && ((JValue)value).Type == JTokenType.String)
-                        {
-                            // convert raw strings so they get serialized properly below
-                            content = (string)value;
-                        }
+                // TODO: Improve this logic
+                // Sniff the object to see if it looks like a response object
+                // by convention
+                object value = null;
+                if (responseObject.TryGetValue("body", out value))
+                {
+                    // the response content becomes the specified body value
+                    content = value;
 
-                        if (jo.TryGetValue("headers", StringComparison.OrdinalIgnoreCase, out value) && value is JObject)
-                        {
-                            headers = (JObject)value;
-                        }
+                    IDictionary<string, object> headersValue = null;
+                    if (responseObject.TryGetValue<IDictionary<string, object>>("headers", out headersValue))
+                    {
+                        responseHeaders = headersValue.AsCaseInsensitive();
+                    }
 
-                        if ((jo.TryGetValue("status", StringComparison.OrdinalIgnoreCase, out value) && value is JValue) ||
-                            (jo.TryGetValue("statusCode", StringComparison.OrdinalIgnoreCase, out value) && value is JValue))
-                        {
-                            statusCode = (HttpStatusCode)(int)value;
-                        }
+                    int statusValue;
+                    if (responseObject.TryGetValue<int>("status", out statusValue) || 
```

Comment:
pretty sure we need to support string for this as well :/ I remember some tests breaking when I treated status as an integer